fastly/block_fastly_service_v1_backend.go

```diff
@@ -309,11 +319,14 @@ func flattenBackends(backendList []*gofastly.Backend) []map[string]interface{} {
 			"ssl_cert_hostname":     b.SSLCertHostname,
 			"ssl_sni_hostname":      b.SSLSNIHostname,
 			"weight":                int(b.Weight),
-			"request_condition":     b.RequestCondition,
 			"healthcheck":           b.HealthCheck,
 		}
 
-		bl = append(bl, nb)
+		if sa.serviceType == ServiceTypeVCL {
+			backend["request_condition"] = b.RequestCondition
+		}
```

Comment:
Is there a way that we could centralize this `serviceType` logic?

For example, maybe a function that returns an object with the fields populated and `RequestCondition` either gets a value or not. And then, in functions similar to this one, we could safely just do `"request_condition" = somereturnval.RequestCondition`

(Just a thought. **Definitely not a blocker**. But if you can think of a nice way to centralize this, awesome. If it is just a hack, I think I would choose what you have since it is pretty clear.)

Reply:
My thinking was this is good for now and relatively clear. If another serviceType comes along then we should certainly discuss more abstraction so we don't descend into conditional hell. 

Not dogmatic about it though.